File: javascript/Hash/algoLadderStarter/mostFrequentLetter/mostFrequentLetter.js
// Given a string, find the most commonly occurring letter.

// Input: “peter piper picked a peck of pickled peppers”
// Output: “p”

const mostFrequentLetter = function (string) {
  let parsedString = string.split("")
  let letterCounts = {}
  let mostFrequentLetterVariable = ""
  let mostFrequentCount = 0

  for (let i = 0; i < parsedString.length; i += 1) {
    if (!letterCounts[parsedString[i]]) {
      letterCounts[parsedString[i]] = 1
    } else {
      letterCounts[parsedString[i]] += 1
    }
    if (letterCounts[string[i]] > mostFrequentCount) {
      mostFrequentCount = letterCounts[string[i]]
      mostFrequentLetterVariable = string[i]
    }
  }
  
  return mostFrequentLetterVariable
}

console.log(mostFrequentLetter("peter piper picked a peck of pickled peppers"))

File: ruby/algo_ladder/most_frequent_letter/most_frequent_letter.rb
# Given a string, find the most commonly occurring letter.

# Input: “peter piper picked a peck of pickled peppers”
# Output: “p”

def most_frequent_letter(input_string)
  most_frequent_letter = ""
  most_frequent_letter_count = 0
  no_spaced_string = input_string.delete(" ")
  letter_hash = {}
  index_one = 0
  while index_one < no_spaced_string.length
    if !letter_hash[no_spaced_string[index_one]]
      letter_hash[no_spaced_string[index_one]] = 1
    else
      letter_hash[no_spaced_string[index_one]] += 1
    end

    if letter_hash[no_spaced_string[index_one]] > most_frequent_letter_count
      most_frequent_letter = no_spaced_string[index_one]
      most_frequent_letter_count = letter_hash[no_spaced_string[index_one]]
    end
    index_one += 1
  end
  return most_frequent_letter
end

p most_frequent_letter("peter piper picked a peck of pickled peppers")